test/e2e/utils/tls.go

```diff
@@ -0,0 +1,250 @@
+		},
+		RootCAs:    certpool,
+		ServerName: ServerName,
+	}
+	return
+}
+
+func ca() *certHolder {
+	// create our private and public key
+	privateKey, err := rsa.GenerateKey(rand.Reader, keyBits)
+	ExpectNoError(err)
+
+	// create the CA
+	ca := &x509.Certificate{
+		SerialNumber: big.NewInt(2019),
+		Subject: pkix.Name{
+			CommonName:         "CA",
+			Organization:       []string{"JBoss"},
+			OrganizationalUnit: []string{"Infinispan"},
+			Locality:           []string{"Red Hat"},
+		},
+		NotBefore:             time.Now(),
+		NotAfter:              time.Now().AddDate(10, 0, 0),
+		IsCA:                  true,
+		BasicConstraintsValid: true,
+		PublicKeyAlgorithm:    x509.RSA,
+		SignatureAlgorithm:    x509.SHA256WithRSA,
+	}
+	certBytes, err := x509.CreateCertificate(rand.Reader, ca, ca, &privateKey.PublicKey, privateKey)
+	ExpectNoError(err)
+
+	cert, err := x509.ParseCertificate(certBytes)
+	ExpectNoError(err)
+
+	return &certHolder{
+		privateKey: privateKey,
+		cert:       cert,
+		certBytes:  certBytes,
+	}
+}
+
+func cert(name string, ca *certHolder) *certHolder {
+	// create our private and public key
+	privateKey, err := rsa.GenerateKey(rand.Reader, keyBits)
+	ExpectNoError(err)
+
+	// set up our server certificate
+	server := &x509.Certificate{
+		SerialNumber: big.NewInt(serialNumber),
+		Subject: pkix.Name{
+			CommonName:         name,
+			Organization:       []string{"JBoss"},
+			OrganizationalUnit: []string{"Infinispan"},
+			Locality:           []string{"Red Hat"},
+		},
+		Issuer:             ca.cert.Subject,
+		NotBefore:          time.Now(),
+		NotAfter:           time.Now().AddDate(10, 0, 0),
+		PublicKeyAlgorithm: x509.RSA,
+		SignatureAlgorithm: x509.SHA256WithRSA,
+	}
+	serialNumber++
+
+	certBytes, err := x509.CreateCertificate(rand.Reader, server, ca.cert, &privateKey.PublicKey, ca.privateKey)
+	ExpectNoError(err)
+
+	cert, err := x509.ParseCertificate(certBytes)
+	ExpectNoError(err)
+
+	return &certHolder{
+		privateKey: privateKey,
+		cert:       cert,
+		certBytes:  certBytes,
+	}
+}
+
+func createKeystore(ca, server *certHolder) []byte {
+	// It's not possible to use the p12 library as we get the following error
+	// TLS handshake failed: javax.net.ssl.SSLException: error:1417A0C1:SSL routines:tls_post_process_client_hello:no shared cipher
+	// keystore, err := p12.Encode(rand.Reader, server.privateKey, server.cert, []*x509.Certificate{ca.cert}, KeystorePassword)
+	var fileMode os.FileMode = 0777
+	ExpectNoError(os.MkdirAll(tmpDir, fileMode))
+	defer os.RemoveAll(tmpDir)
+
+	privKeyFile := tmpFile("server_key.pem")
+	certFile := tmpFile("server_cert.pem")
+	keystorefile := tmpFile("keystore.p12")
+
+	err := ioutil.WriteFile(privKeyFile, server.getPrivateKeyPEM(), fileMode)
+	ExpectNoError(err)
+
+	err = ioutil.WriteFile(certFile, append(server.getCertPEM(), ca.getCertPEM()...), fileMode)
+	ExpectNoError(err)
+
+	cmd := exec.Command("openssl", "pkcs12", "-export", "-in", certFile, "-inkey", privKeyFile,
```

Comment:
We should ensure that `openssl` command will work similar on all testing platforms (don't forget about IBM colleagues).
Otherwise this command should be running inside container to be fully compatible (preferred way)

Reply:
`openssl` is available on most platforms (including z), so I don't think we have to worry too much. Plus, the IBM team actually execute the tests on a x64 machine against a z k8s cluster.

> Otherwise this command should be running inside container to be fully compatible (preferred way)

This isn't so simple as we need to read the created files, so it would require using volumes etc.